Ensure Ubuntu builds can find passenger_native_support.so

treydock opened the pull request:
Disable attemped download of passenger native support binary
Fixes #2051

In a testing container I removed the nginx_stage profile changes and verified the download was not attempted:

```
App 7191 output:  [passenger_native_support.so] not found for current Ruby interpreter.
App 7191 output:      This library provides various optimized routines that make
App 7191 output:      Phusion Passenger(R) faster. Please run 'sudo yum install passenger-devel-6.0.14'
App 7191 output:      so that Phusion Passenger(R) can compile one on the next run.
App 7191 output:  [passenger_native_support.so] not downloading because PASSENGER_DOWNLOAD_NATIVE_SUPPORT_BINARY=0
App 7191 output:  [passenger_native_support.so] will not be used (can't compile or download) 
App 7191 output:   --> Passenger will still operate normally.
```

WIth all the changes in pull request, all output about the passenger_native_support.so is gone when testing with Ubuntu 22.04.



┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202353833422339) by [Unito](https://www.unito.io)

treydock commented on nginx_stage/etc/profile:
```diff
@@ -8,4 +8,9 @@
+  source "${SCL_SOURCE}" enable ${SCL_PKGS}
+else
+  export GEM_PATH="/opt/ood/gems:$GEM_PATH"
+  export RUBYLIB="/opt/ood/ondemand/root/usr/share/ruby/vendor_ruby:${RUBYLIB:+:${RUBYLIB}}"
```
This is normally set by the SCL enable code so this is to ensure proper paths for Ubuntu.

I wonder though if should instead create like `/opt/ood/profile` with these items in it for just Ubuntu and source that file.  Could be sourced by other profiles if ever became needed.  Could go so far as to mimic SCL and create `opt/ood/ondemand/enable` which is what SCL uses to setup the environment.

johrstrom replied:
> I wonder though if should instead create like /opt/ood/profile

Likely. ood-portal-generator has similar logic. I'd be up for one, doesn't have to be this PR.

https://github.com/OSC/ondemand/blob/master/ood-portal-generator/etc/profile

I think the documentation would read the same, just a different source file.

https://osc.github.io/ood-documentation/latest/customization_overview.html?highlight=etc%20ood%20profile

treydock replied:
The `/etc/ood/profile` is for admin defined values.  I'll go ahead and make the necessary adjustment in this PR, as I had the changes staged just didn't commit yet.

johrstrom reviewed (dismissed):
odd there's no configuration for this.

treydock commented:
@johrstrom Moved the env vars to `/opt/ood/ondemand/enable` for Ubuntu to mimic SCL.  Updated ood-portal-generator and nginx_stage to source that file.  Also added acceptance test for new file.  Ready for new review.

johrstrom commented on packaging/deb/rules:
```diff
@@ -53,6 +53,8 @@ override_dh_auto_install:
 	install -m 644 ood-portal-generator/share/need_auth.html $(APACHE_DIR)/public/need_auth.html
 	install -m 644 hooks/hook.env.example $(CONFIG_DIR)/hook.env
+	mkdir -p $(DESTDIR)/opt/ood/ondemand
+	install -m 644 packaging/files/profile.deb $(DESTDIR)/opt/ood/ondemand/enable
```
Shouldn't we keep the profile name? Or this is just for Ubuntu? 

I guess I'm worried about the toggle in the documentation. Do this for X and this for Y. 

I guess I just assumed what you'd thought was a singe `/opt/ood/profile` that both nginx stage and ood-portal-generator source and that folks can override in /etc/ood/profile.

treydock replied:
I don't think we need to document this file, we don't document the same file for SCL platforms.  This is a file that if modified, package updates will overwrite.  If you'd prefer I can move it to `/opt/ood/profile`, the main motivation for this path was to have Ubuntu use same file as EL platforms which use SCL.

johrstrom replied:
We do have documentation for `/opt/ood/nginx_stage/etc/profile` which on Ubuntu would need to change to `/opt/ood/ondemand/enable`. Or should the docs point to /opt/ood/ondemand/enable instead of /opt/ood/nginx_stage/etc/profile?

https://github.com/OSC/ood-documentation/blob/0a5fe4babd2f60acd678d169b48b9b7958f21320/source/reference/pun-environment.rst?plain=1#L36

My concern is doing different things on different platforms can be confusing not only for folks to implement for us to explain.

johrstrom replied:
OK I think this conversation just illustrates how easily this can get confusing.

It seems like I need a minute to look this over in more detail and that the docs would still hold.

johrstrom commented on packaging/files/profile.deb:
```diff
@@ -0,0 +1,2 @@
+export GEM_PATH="/opt/ood/gems:${GEM_PATH:+:${GEM_PATH}}"
```
Laughably, I think my confusion in the other thread is the name of this file is `profile` implying that it replaced a profile file.

Can we rename this to `enable.deb`?

treydock replied:
I renamed the file to `enable.deb`

johrstrom reviewed (approved):
Thanks!
